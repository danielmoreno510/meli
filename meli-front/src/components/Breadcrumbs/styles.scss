@import '../../index';

.breadcrumbs{
  background-color: $background-color;
  padding: 18px 160px 0px 160px;
  display: flex;
}

.breadcrumbContainer{
  display: flex;
  margin-right: 14px;
}