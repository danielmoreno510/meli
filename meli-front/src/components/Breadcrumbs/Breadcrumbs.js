import React from "react";
import PropTypes from "prop-types";

import "./styles.scss";

const Breadcrumbs = ({ breadcrumbs }) =>
  breadcrumbs.length > 0 ? (
    <div className="breadcrumbs">
      {breadcrumbs.map((breadcrumb, index) => (
        <div className="breadcrumbContainer" key={breadcrumb}>
          <div>{breadcrumb}</div>
          {breadcrumbs.length - 1 !== index && (
            <div>&nbsp;&nbsp;&nbsp;&#62;</div>
          )}
        </div>
      ))}
    </div>
  ) : (
    <div></div>
  );

Breadcrumbs.propTypes = {
  breadcrumbs: PropTypes.array,
};

export default Breadcrumbs;
