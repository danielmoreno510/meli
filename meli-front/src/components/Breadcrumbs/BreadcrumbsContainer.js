import Breadcrumbs from "./Breadcrumbs";
import { StoreService } from "../../state/StoreService.js";

export const mapStateToProps = (state) => ({
  breadcrumbs: state.products.breadcrumbs,
});
// const mapDispatchToProps = (dispatch) => {
//   return {
//   };
// };

const BreadcrumbsContainer = StoreService.connect(
  mapStateToProps,
  // mapDispatchToProps
)(Breadcrumbs);
export default BreadcrumbsContainer;
