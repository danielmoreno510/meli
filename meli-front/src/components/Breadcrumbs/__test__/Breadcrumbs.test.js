import React from "react";
import { shallow } from "enzyme";

import Breadcrumbs from "../Breadcrumbs";

describe("Breadcrumbs component", () => {
  const testProps = (breadcrumbs) => ({
    breadcrumbs,
  })

  describe("WHEN the Breadcrumbs component is render", () => {
    const wrapper = shallow(<Breadcrumbs {...testProps(["test", "test"])} />);

    it("THEN should display Breadcrumbs component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("WHEN the Breadcrumbs component is render without props", () => {
    const wrapper = shallow(<Breadcrumbs {...testProps([])} />);

    it("THEN should display Breadcrumbs component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
