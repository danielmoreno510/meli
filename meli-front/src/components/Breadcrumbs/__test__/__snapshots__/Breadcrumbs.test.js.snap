// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component WHEN the Breadcrumbs component is render THEN should display Breadcrumbs component 1`] = `
<div
  className="breadcrumbs"
>
  <div
    className="breadcrumbContainer"
    key="test"
  >
    <div>
      test
    </div>
    <div>
         &gt;
    </div>
  </div>
  <div
    className="breadcrumbContainer"
    key="test"
  >
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`Breadcrumbs component WHEN the Breadcrumbs component is render without props THEN should display Breadcrumbs component 1`] = `<div />`;
